### PR TITLE
Start migration of Gradle from Groovy to Kotlin

### PR DESCRIPTION
**Pull request by geekygecko**
## Description

This change follows the Google recommendation of migrating the Gradle scripts to Kotlin. 

> Kotlin is preferred over the Groovy for writing Gradle scripts because Kotlin is more readable and offers better compile-time checking and IDE support.

Unfortunately, I couldn't migrate `base.gradle` due to an error. Hopefully, I can figure out how to do this in a future PR. https://docs.gradle.org/current/userguide/kotlin_dsl.html#type-safe-accessors

```
e: file:///.../pocket-casts-android/base.gradle.kts:15:1: Unresolved reference: android
```

### Check all the build variants assemble
1. Build all variants `./gradlew clean assemble`.
2. ✅ Confirm there are apks in the following location:
- app/build/outputs/apk/debug/
- app/build/outputs/apk/debugProd/
- app/build/outputs/apk/release/
- automotive/build/outputs/apk/debug/
- automotive/build/outputs/apk/debugProd/
- automotive/build/outputs/apk/release/
- wear/build/outputs/apk/debug/
- wear/build/outputs/apk/debugProd/
- wear/build/outputs/apk/release/
3. ✅ Verify the release mobile app runs

### Check a release build without signing information
1. Comment out any signing information in any file you might have in `secret.properties`, `local.properties`, or `~/.gradle/gradle.properties`.
4. Build the app `./gradlew clean assembleRelease`
5. ✅ Confirm an unsigned apk has been created. `app/build/outputs/apk/release/app-release-unsigned.apk`.

### Check a release build with signing information
1. Add signing information into `secret.properties`.
```
signingKeyStoreFile=
signingKeyStorePassword=
signingKeyAlias=
signingKeyPassword=
```
3. Build the app `./gradlew clean assembleRelease`
6. ✅ Confirm a signed apk has been created. `app/build/outputs/apk/release/app-release.apk`.

### Check a release build with signing information in Gradle properties
1. Comment out the signing information in `secret.properties`.
2. Add signing information into `~/.gradle/gradle.properties`.
```
pocketcastsKeyStoreFile=
pocketcastsKeyStorePassword=
pocketcastsKeyStoreAlias=
pocketcastsKeyStoreAliasPassword=
```
3. Build the app `./gradlew clean assembleRelease`
4. ✅ Confirm a signed apk has been created. `app/build/outputs/apk/release/app-release.apk`.

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -1,115 +0,0 @@
...
-    buildTypes {
-        debug {
-            applicationIdSuffix '.debug'
-
-            def gitLongVersion = "git -C ${rootDir} describe --tags --long --dirty".execute().text.trim()
-            def (gitTag, versionBuild, gitHash, gitDirty) = gitLongVersion.tokenize('-')
-
-            manifestPlaceholders = [
-                    appIcon: "@mipmap/ic_launcher_radioactive",
-                    gitHash: gitHash + (gitDirty ? ("-" + gitDirty) : ""),
```
Comment (geekygecko) [inline app/build.gradle:45]: I have removed the git hash as we no longer use it on the about page.

File: dependencies.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+        // Android
+        set("minSdkVersion", 23)
+        set("minSdkVersionWear", 26)
+        set("targetSdkVersion", 33)
+        set("compileSdkVersion", 33)
+        set("testInstrumentationRunner", "androidx.test.runner.AndroidJUnitRunner")
+
+        // App Signing
+        val storePath = secretProperties.getProperty("signingKeyStoreFile", null)
+        val storeFile = file(storePath)
```
Comment (geekygecko) [inline dependencies.gradle.kts:68]: The old line seemed to work with absolute and relative paths. It had broken for my absolute path. 
```
storeFile = file("$rootDir/${secretProperties.getProperty("signingKeyStoreFile", null)}")
```
The `file` command documentation says the following so we don't require the `rootDir`. I have test this and it seems to work fine.

> Resolves a file path relative to the project directory of this project. This method converts the supplied path based on its type: 

https://docs.gradle.org/current/dsl/org.gradle.api.Project.html#org.gradle.api.Project:file(java.lang.Object)

File: dependencies.gradle.kts [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+        // Next check Gradle properties such as local.properties or ~/.gradle/gradle.properties
+        if (!canSignRelease && project.hasProperty("pocketcastsKeyStoreFile")) {
+            val pocketcastsKeyStoreFile = project.property("pocketcastsKeyStoreFile").toString()
+            if (!pocketcastsKeyStoreFile.isNullOrBlank()) {
+                val storeFile = file(pocketcastsKeyStoreFile)
+                canSignRelease = storeFile.exists()
+                set("storeFile", storeFile)
+                set("storePassword", project.property("pocketcastsKeyStorePassword").toString())
+                set("keyAlias", project.property("pocketcastsKeyStoreAlias").toString())
+                set("keyPassword", project.property("pocketcastsKeyStoreAliasPassword").toString())
```
Comment (mchowning) [inline dependencies.gradle.kts:83]: Not for this PR, but do you think there's any reason we couldn't change these keys to match the keys we use in `secrets.properties`?

**Review by mchowning — APPROVED** [review]
This looks great and is working perfectly in my testing! Thanks so much for tackling this! 🤩

File: dependencies.gradle.kts (hunk 3 above)
Comment (geekygecko) [reply dependencies.gradle.kts:83]: We used to have Pocket Weather too so I had them both stored in my `~/.gradle/gradle.properties` file. Good idea, I will change this.

**Comment by pocketcasts** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">PR requires at least one label.</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 500 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">PR is not assigned to a milestone.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by pocketcasts** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 500 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">PR is not assigned to a milestone.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Pull request merged**